### PR TITLE
feat(csv): adding data model to csv analyzer

### PR DESCRIPTION
**Pull request by neubs-bsi**
corresponding issue: #193 

Currently Included: 
- Observed License
- Source URL
- Release Artifact URL 
- Software Heritage URL
- coordinate type

Missing:
Due to belonging exclusively to SW360 data model at the moment
- Change Status
- Clearing State

Due to unclear format
- CopyrightStatement

### CODE REVIEW

File: antenna-documentation/src/site/markdown/analyzers/csv-analyzer-step.md [hunk 1]
```diff
@@ -2,8 +2,8 @@
 This analyzer investigates a given *csv* file. The format of this csv file follows this format:
 
 ```
-"Artifact Id","Group Id","Version","License Short Name","License Long Name","File Name"
```
Comment (maxhbr) [inline antenna-documentation/src/site/markdown/analyzers/csv-analyzer-step.md]: Why did you remove the quotes? In CSV they are commonly used.

Comment (maxhbr) [reply antenna-documentation/src/site/markdown/analyzers/csv-analyzer-step.md]: Is this a breaking change or is this still able to parse the "old" CSV files?

File: core/core-workflow-steps/src/main/java/org/eclipse/sw360/antenna/workflow/analyzers/CsvAnalyzer.java [hunk 2]
```diff
@@ -41,6 +42,17 @@
     private static final String LICENSE_SHORT_NAME = "License Short Name";
     private static final String LICENSE_LONG_NAME = "License Long Name";
     private static final String PATH_NAME = "File Name";
+    private static final String COORDINATE_TYPE = "Coordinate Type";
+    private static final String OBSERVED_LICENSE_SHORT_NAME = "Observed License Short Name";
+    private static final String OBSERVED_LICENSE_LONG_NAME = "Observed License Long Name";
+    private static final String SOURCE_URL = "Source URL";
+    private static final String RELEASE_ARTIFACT_URL = "Release Artifact URL";
+    private static final String SWH_URL = "Software Heritage URL";
+//    private static final String ClEARING_STATE = "Clearing State";
```
Comment (maxhbr) [inline core/core-workflow-steps/src/main/java/org/eclipse/sw360/antenna/workflow/analyzers/CsvAnalyzer.java]: commented code

File: core/core-workflow-steps/src/main/java/org/eclipse/sw360/antenna/workflow/analyzers/CsvAnalyzer.java [hunk 3]
```diff
@@ -100,4 +112,34 @@ public WorkflowStepResult yield() throws AntennaException {
...
+
+    private License createLicense(String short_name, String long_name) {
+        License license = new License();
+        license.setName(short_name);
+        license.setLongName(long_name);
+
+        return license;
+    }
+
+    private void mapCoordinate(Artifact artifact, CSVRecord record) {
```
Comment (maxhbr) [inline core/core-workflow-steps/src/main/java/org/eclipse/sw360/antenna/workflow/analyzers/CsvAnalyzer.java]: instead of this style of coding with mutations it would maybe be more consistent to use the same pattern as for `createLicense` e.g. 
```suggestion
    private FactOrSomeOtherMatchingClass mapCoordinate(CSVRecord record) {
```

File: core/core-workflow-steps/src/test/resources/CsvAnalyzerTest/dependencies.csv [hunk 4]
```diff
@@ -1,2 +1,2 @@
-Artifact Id,Group Id,Version,License Short Name,License Long Name,File Name
-commons-csv,org.apache.commons,1.4,Apache-2.0,Apache Software License 2.0,commons-csv.jar
+Artifact Id,Group Id,Version,License Short Name,License Long Name,File Name,Coordinate Type,Observed License Short Name,Observed License Long Name,Source URL,Release Artifact URL,Software Heritage URL
```
Comment (maxhbr) [inline core/core-workflow-steps/src/test/resources/CsvAnalyzerTest/dependencies.csv]: maybe add a test, that the old CSV format is still supported?

File: antenna-documentation/src/site/markdown/analyzers/csv-analyzer-step.md [hunk 5]
```diff
@@ -2,10 +2,14 @@
...
-commons-csv,org.apache.commons,1.4,Apache-2.0,Apache Software License 2.0,commons-csv.jar
+Artifact Id,Group Id,Version,License Short Name,License Long Name,File Name,Coordinate Type,Observed License Short Name,Observed License Long Name,Source URL,Release Artifact URL,Software Heritage URL
+commons-csv,org.apache.commons,1.4,Apache-2.0,Apache Software License 2.0,commons-csv.jar,mvn,Apache-2.0,Apache Software License 2.0,http://archive.apache.org/dist/commons/csv/source/commons-csv-1.4-src.zip,url,url
 ```
 
+Note: In the previous documentation the format was given with the headers of the csv columns in quotation marks.
```
[the remark this answers is not in the document]
Comment (neubs-bsi) [reply antenna-documentation/src/site/markdown/analyzers/csv-analyzer-step.md]: I added the quotation marks again and took the note out

File: core/core-workflow-steps/src/main/java/org/eclipse/sw360/antenna/workflow/analyzers/CsvAnalyzer.java [hunk 6]
```diff
@@ -100,4 +104,45 @@ public WorkflowStepResult yield() throws AntennaException {
...
+                    createLicense(record.get(OBSERVED_LICENSE_SHORT_NAME), record.get(OBSERVED_LICENSE_LONG_NAME))));
+        }
+        if(record.isMapped(SOURCE_URL)) {
+            artifact.addFact(new ArtifactSourceUrl(record.get(SOURCE_URL)));
+        }
+        if(record.isMapped(RELEASE_ARTIFACT_URL)) {
+            artifact.addFact(new ArtifactReleaseTagURL(record.get(RELEASE_ARTIFACT_URL)));
+        }
+        if(record.isMapped(SWH_URL)) {
+                artifact.addFact(new ArtifactSoftwareHeritageURL(record.get(SWH_URL)));
```
Comment (maxhbr) [inline core/core-workflow-steps/src/main/java/org/eclipse/sw360/antenna/workflow/analyzers/CsvAnalyzer.java]: formatting

Comment (neubs-bsi) [reply core/core-workflow-steps/src/main/java/org/eclipse/sw360/antenna/workflow/analyzers/CsvAnalyzer.java]: formatted. Also removed two redundant imports from `CSVAnalyzerTest`

**Comment by neubs-bsi** [comment]
It should work now, but @maxhbr do not merge yet, since I want to find out why MatchState is necessary before merging.

**Comment by neubs-bsi** [comment]
Source of previous failing of Rule Engine has been determined: 
https://github.com/eclipse/antenna/blob/17f055e7728ece7a0cbcc71189394b11962cb3f8/core/model/src/main/java/org/eclipse/sw360/antenna/model/artifact/Artifact.java#L79-L83
sets MatchState default to UNKNOWN and 
assembly/common-rules/src/main/resources/policies/rules/A1_Unknown_Artifact.drl
fails when Matchstate is UNKNOWN.
This is okay. @maxhbr PR can be merged
